.github/workflows/unit.yaml

```diff
@@ -0,0 +1,46 @@
+name: Unit tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9, pypy3]
+        include:
+        - python-version: 3.8
+          publish-coverage: true
```

Comment:
I guess that for pydoctor this works as there is no python specific code.

But you should be able to publish all version to codecov and it will take care of merging then.

Reply:
There are tests that only run on specific Python versions, but those are active in 3.8.

In Travis coverage uploads were also tied to a specific Python version. It seems it was introduced in commit 32fe9bcd, but that was 4 years ago, so I can imagine you don't remember the details anymore.

Reply:
I think that the thing `test-py27-codecov-travis` was done not for coverage but for PyPI publishing.

so this is not about publish coverage, but about publish the PyPI package.

I think that you can remove the whole `include` section.

GitHub Action has a much nicer API and tools and we no longer need that hack.